mbtiles/src/pool.rs

```diff
@@ -6,14 +6,109 @@ use sqlx::{Pool, Sqlite, SqlitePool};
 use crate::errors::MbtResult;
 use crate::{MbtType, Mbtiles, Metadata};
 
+/// Connection pool for concurrent read access to an `MBTiles` file.
+///
+/// `MbtilesPool` wraps an [`Mbtiles`] reference with a `SQLite` connection pool,
+/// enabling safe concurrent access for tile serving applications. This is the
+/// recommended type for production tile servers.
+///
+/// # Connection Pooling
+///
+/// The pool manages multiple `SQLite` connections to the same file, allowing
+/// concurrent read operations without blocking. This is particularly useful
+/// for web servers handling multiple simultaneous tile requests.
+///
+/// # Examples
+///
+/// ## Basic tile serving
+///
+/// ```
+/// use mbtiles::MbtilesPool;
+///
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// // Open file with connection pool
+/// let pool = MbtilesPool::open_readonly("world.mbtiles").await?;
+///
+/// // Get metadata (automatically acquires connection from pool)
+/// let metadata = pool.get_metadata().await?;
+/// println!("Tileset: {}", metadata.tilejson.name.unwrap_or_default());
+///
+/// // Fetch a tile - connection is automatically managed
+/// if let Some(tile_data) = pool.get_tile(4, 5, 6).await? {
+///     println!("Tile size: {} bytes", tile_data.len());
+/// }
+/// # Ok(())
+/// # }
+/// ```
+///
+/// ## Concurrent tile requests
+///
+/// ```
+/// use mbtiles::MbtilesPool;
+/// use std::sync::Arc;
+///
+/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+/// let pool = Arc::new(MbtilesPool::open_readonly("world.mbtiles").await?);
+///
+/// // Spawn multiple concurrent tile requests
+/// let mut handles = vec![];
+/// for z in 0..5 {
+///     let pool = Arc::clone(&pool);
+///     handles.push(tokio::spawn(async move {
+///         pool.get_tile(z, 0, 0).await
+///     }));
+/// }
+///
+/// // All requests run concurrently using different pool connections
+/// for handle in handles {
+///     let result = handle.await??;
+///     println!("Got tile: {:?}", result.is_some());
+/// }
+/// # Ok(())
+/// # }
```

Comment:
No need to Arc. We already do this intenally 😉 


```suggestion
///
/// # async fn example() -> Result<(), Box<dyn std::error::Error>> {
/// let pool = MbtilesPool::open_readonly("world.mbtiles").await?;
///
/// // Spawn multiple concurrent tile requests
/// let mut handles = vec![];
/// for z in 0..5 {
///     // cheap and thead-save
///     let pool = pool.clone();
///     handles.push(tokio::spawn(async move {
///         pool.get_tile(z, 0, 0).await
///     }));
/// }
///
/// // All requests run concurrently using different pool connections
/// for handle in handles {
///     let result = handle.await??;
///     println!("Got tile: {:?}", result.is_some());
/// }
/// # Ok(())
/// # }
```